EngineLayer/CrosslinkSearch/CrosslinkSpectralMatch.cs

```diff
@@ -87,31 +87,12 @@ public static List<int> GetPossibleCrosslinkerModSites(char[] crosslinkerModSite
             List<int> possibleXlPositions = new List<int>();
             bool wildcard = crosslinkerModSites.Any(p => p == 'X');
 
-            //Consider the possibility that the site is at Protein N terminal. 
-            if (crosslinkerModSites.Contains(peptide.BaseSequence[0]))
-            {
-                if (!CrosslinkAtCleavageSite)
-                {
-                    if (peptide.OneBasedStartResidueInProtein == 1
-                || (peptide.OneBasedStartResidueInProtein == 2 && initiatorMethionineBehavior != InitiatorMethionineBehavior.Retain)
-                || peptide.Protein.ProteolysisProducts.Any(x => x.OneBasedBeginPosition == peptide.OneBasedStartResidueInProtein))
-                    {
-                        possibleXlPositions.Add(1);
-                    }
-                }
-                else
-                {
-                    possibleXlPositions.Add(1);
-                }
-            }
-
-
-            List<int> range = Enumerable.Range(1, peptide.BaseSequence.Length - 1).ToList();
+            List<int> range = Enumerable.Range(0, peptide.BaseSequence.Length).ToList();
             if (!CrosslinkAtCleavageSite && peptide.OneBasedEndResidueInProtein != peptide.Protein.Length 
                 && !peptide.Protein.ProteolysisProducts.Any(x => x.OneBasedEndPosition == peptide.OneBasedEndResidueInProtein))
             {
-                //The N terminal and C termial cannot be crosslinked and cleaved.
-                range = Enumerable.Range(1, peptide.BaseSequence.Length - 2).ToList();
+                //The C termial cannot be crosslinked and cleaved.
+                range = Enumerable.Range(0, peptide.BaseSequence.Length - 1).ToList();
```

Comment:
Want a unit test the allows AspN to be crosslinked at the C-terminus, but not at the N-terminus. The current handling appears trypsin/LysC/ArgC-specific.

Reply:
I think this is fine for right now, but it might crop up as an issue down the line. It depends on the protease mechanism, so I'm not sure we can predict this.